specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -4383,6 +4383,13 @@
             "url": "https://docs.microsoft.com/rest/api/searchservice/Custom-analyzers-in-Azure-Search"
           }
         },
+        "encryptionKey": {
+          "$ref": "#/definitions/EncryptionKey",
+          "description": "The configuration to use when you want to encrypt the index with your own encryption key (customer-managed key). On updates, the encryption key configuration will stay unchanged if not specified or null. Encryption with customer-managed keys is not available for free services. For paid services, it is only available for services created on or after 2019-01-1.",
```

Comment:
Thanks for the reformulation. A few points:
- user-managed encryption is for both indexes and synonym maps
- When used on indexes, it applies to all fields. There's no way to selectively decide on which field to apply encryption
- technically, I don't think I would consider encryption with CMK as overriding the default encryption, it mostly adds a layer on top. The default "disk-level" encryption is still turned on, but we also add a second layer of encryption at the index file level.
- I also noticed you removed the sentence: " On updates, the encryption key configuration will stay unchanged if not specified or null". Do you think we should do without it? Since this is a pretty unique behavior of that property, I feel like it should be noted somewhere in the documentation. The Storage team did decide to also include that detail in their own EncryptionKey description.

Here's my suggested edit:
"Definition of a fully-specified user-managed encryption key, created in Azure Key Vault, and used for data-at-rest encryption of indexes and synonym maps in Azure Search. On updates, the definition will stay unchanged if not specified or null. User-managed keys are a preview feature, available to billable search services created on or after January 1 2019."

Reply:
WRT bullet 2, I always that it was just a key thing -- our keys or your keys. But I see now that its for people who need assurance that MSFT can't view or access their data on the backend.   What about this:

"Adds a private encryption layer over indexes in Azure Search. User-managed encryption is in addition to Microsoft-managed encryption, and used when you want full assurance that no one can access, retrieve, or restore your keys, or decrypt your data on the backend. User-managed keys are a preview feature, available to billable search services created on or after January 1 2019. You must Azure Key Vault for key definition, access policies, and oversight. Once you add an encryption definition, it stays with index on future updates as long as you do not overwrite it or replace with nulls."

The 'update' is a can of worms.  I wonder if we should just link out to Nati's doc.  If I overwrite the definition on an index that is already encrypted, I can't decrypt it ever again, can I?

Reply:
@HeidiSteen Some feedback on your proposed text:
- The updated text seems broader than it needs to be, especially the part about "access, retrieve, or restore your keys".
- The update scenario is not as bad as it sounds, and I think we need to explain it clearly here.
- Something to keep in mind with Swagger is that each description has "Gets or sets" tacked in front of it in generated SDK comments, so starting with "Adds" doesn't make sense. We basically have to define properties in Swagger as nouns, starting with an article.
- No need to call out that this is a preview feature in the Swagger -- the entire Swagger definition is for a preview API version.

@shmed Regarding update, we've been focusing on what happens when you pass null or omit the encryption key. What about key rotation? That is allowed, right (since this key is just encrypting the DEK, which never changes)? Assuming that is the case, here's my proposed revision:

"A description of an encryption key that you create in Azure Key Vault. This key is used to provide an additional level of encryption-at-rest for your data when you want full assurance that no one, not even Microsoft, can decrypt your data in Azure Search. Once you have encrypted your data, it will always remain encrypted. Azure Search will ignore attempts to set this property to null. You can change this property as needed if you want to rotate your encryption key; Your data will be unaffected. Encryption with user-managed keys is not available for free search services, and is only available for paid services created on or after January 1, 2019."

Reply:
@brjohnstmsft Thanks for stepping in with the edit work.  I'm struggling with my own learning curve on this feature so I'm not quite able to help out like I should.  

I like the latest version for the description on line 4388.  This statement, "Azure Search will ignore attempts to set this property to null", and that ones after it, are very helpful.